src/section/help.rs

```diff
@@ -1,10 +1,76 @@
-use crate::{Report, Result};
+//! Provides an extension trait for attaching `Section`s to error reports.
+use crate::{section, Report, Result, Section};
 use ansi_term::Color::*;
 use std::fmt::{self, Display};
 
 /// A helper trait for attaching help text to errors to be displayed after the chain of errors
+///
+/// `color_eyre` provides two types of help text that can be attached to error reports, custom
+/// sections and pre-configured sections. Custom sections are added via the `section` and
+/// `with_section` methods, and give maximum control over formatting. For more details check out
+/// the docs for [`Section`].
+///
+/// The pre-configured sections are provided via `suggestion`, `warning`, and `note`. These
+/// sections are displayed after all other sections with no extra newlines between subsequent Help
+/// sections. They consist only of a header portion and are prepended with a colored string
+/// indicating the kind of section, e.g. `Note: This might have failed due to ..."
+///
+/// [`Section`]: struct.Section.html
 pub trait Help<T>: private::Sealed {
-    /// Add a note to an error, to be displayed after the chain of errors.
+    /// Add a section to an error report, to be displayed after the chain of errors.
+    ///
+    /// Sections are displayed in the order they are added to the error report. They are displayed
+    /// immediately after the `Error:` section and before the `SpanTrace` and `Backtrace` sections.
+    /// They consist of a header and an optional body. The body of the section is indented by
+    /// default.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,should_panic
+    /// use color_eyre::{Report, Help};
+    /// use eyre::eyre;
+    ///
+    /// Err(eyre!("command failed"))
+    ///     .section("Please report bugs to https://real.url/bugs")?;
+    /// # Ok::<_, Report>(())
+    /// ```
+    fn section<C>(self, section: C) -> Result<T>
+    where
+        C: Into<Section>;
+
+    /// Add a section to an error report, to be displayed after the chain of errors, which is
+    /// lazily evaluated only in the case of an error
```

Comment:
nit, consistency: you've capitalised the Thing which gets added to the error report further down this file. Also elsewhere you've had a full stop.

```suggestion
    /// Add a Section to an error report, to be displayed after the chain of errors, which is
    /// lazily evaluated only in the case of an error.
```

Is it the Section or the chain of errors that gets lazily evaluated? I think it's the former, but this sentence doesn't make that entirely clear.

Reply:
It's the section, I can see the confusion though and I'll reword this